Kickstarter-iOS/DataSources/LiveStreamDiscoveryDataSource.swift

```diff
@@ -0,0 +1,91 @@
+import Library
+import LiveStream
+import Prelude
+import UIKit
+
+internal final class LiveStreamDiscoveryDataSource: ValueCellDataSource {
+  internal func load(liveStreams: [LiveStreamEvent]) {
+    self.clearValues()
+
+    sorted(liveStreamEvents: liveStreams)
+      .groupedBy(sectionTitle(forLiveStreamEvent:))
+      .forEach { title, events in
+        guard !events.isEmpty else { return }
+        self.appendRow(value: title, cellClass: LiveStreamDiscoveryTitleCell.self, toSection: title.section)
+        events.forEach { event in
+          self.appendRow(value: event, cellClass: LiveStreamDiscoveryCell.self, toSection: title.section)
+        }
+    }
+  }
+
+  internal override func configureCell(tableCell cell: UITableViewCell, withValue value: Any) {
+    switch (cell, value) {
+    case let (cell as LiveStreamDiscoveryCell, value as LiveStreamEvent):
+      cell.configureWith(value: value)
+    case let (cell as LiveStreamDiscoveryTitleCell, value as LiveStreamDiscoveryTitleType):
+      cell.configureWith(value: value)
+    default:
+      assertionFailure("Unrecognized combo: \(cell), \(value)")
+    }
+  }
+}
+
+private func sorted(liveStreamEvents: [LiveStreamEvent]) -> [LiveStreamEvent] {
```

Comment:
if the code in this function looks suspiciously like one i cooked up in #45, then you are right. unfortunately i cant share the code cause one operates on `LiveStreamEvent`s and the other on `Project.LiveStream` :(((

the models are a bit messy due to hitting two different apis right now. we are gonna try to find a way to bridge these two worlds soon.